test/GitHub.App.UnitTests/Services/RepositoryCloneServiceTests.cs

```diff
@@ -122,6 +122,23 @@ public async Task UpdatesMetricsWhenDefaultClonePath(string targetPath, string d
                     ((MemberExpression)x.Body).Member.Name == counterName));
         }
 
+        [Test]
+        public async Task CleansDirectoryOnCloneFailed()
+        {
+            var serviceProvider = Substitutes.GetServiceProvider();
+            var operatingSystem = serviceProvider.GetOperatingSystem();
+            var vsGitServices = serviceProvider.GetVSGitServices();
+            var cloneService = CreateRepositoryCloneService(serviceProvider);
+
+            Assert.ThrowsAsync<Exception>(async () => {
```

Comment:
This could be simplified to just:
```cs
Assert.ThrowsAsync<Exception>(() => cloneService.CloneRepository("https://github.com/failing/url", @"c:\dev\bar"));
```

Reply:
Thanks. 😄